test/com/moclojer/rq/queue_test.clj

```diff
@@ -0,0 +1,27 @@
+(ns com.moclojer.rq.queue-test
+  (:require
+   [clojure.test :as t]
+   [com.moclojer.rq :as rq]
+   [com.moclojer.rq.queue :as rq-queue]
+   [com.moclojer.test-utils :as utils]))
+
+(t/deftest queue-test
+  (let [client (rq/create-client "redis://localhost:6379")
+        queue-name (str (random-uuid))
+        message (utils/gen-message)]
+
+    (t/testing "raw"
+      (rq-queue/push! client queue-name message)
+      (rq-queue/push! client queue-name (utils/gen-message))
+      (t/is (= 2 (rq-queue/llen client queue-name)))
+      (t/is (= message (rq-queue/pop! client queue-name :direction :r))))
+
+    (t/testing "direction"
+      (rq-queue/push! client queue-name message :direction :r)
+      (t/is (= message (rq-queue/pop! client queue-name :direction :r))))
+
+    (t/testing "pattern"
+      (rq-queue/push! client queue-name message :pattern :pending)
+      (t/is (= message (rq-queue/pop! client queue-name :pattern :pending))))
+
+    (rq/close-client client)))
```

Comment:
@coderabbitai As I said before, we don't need a try catch block in this, since our close-client is only needed when everything goes right with the client. If there's an exception, we are OK with not closing the client, because it is definitely already broken, and closed therefore.